### ISSUE TITLE
[critical] Fail to debug with java 23 - Unrecognized option: -Xnoagent

### ISSUE THREAD

**Issue by gayanper** [microsoft/java-debug#561]
When trying to debug a program using java 23 EA version I get the following error

```
penJDK 64-Bit Server VM warning: Option -Xdebug was deprecated in JDK 22 and will likely be removed in a future release.
Unrecognized option: -Xnoagent
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

**Comment by gayanper** [comment]
@testforstephen Seems like this is cause by the line https://github.com/microsoft/java-debug/blob/e06a1b686625b28a66e0bd6a22f4668fb4f02b4d/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/AdvancedLaunchingConnector.java#L198

**Comment by testforstephen** [comment]
thanks for the reminder. We reference this parameters  `-Xdebug` and `-Xnoagent`  from the old JDT source code, it's time to remove them since they're deprecated.

**Comment by jochen-lise** [comment]
Hello, i get the same error today, trying to debug some tests

```
OpenJDK 64-Bit Server VM warning: Option -Xdebug was deprecated in JDK 22 and will likely be removed in a future release.
Unrecognized option: -Xnoagent
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

**Comment by testforstephen** [comment]
@jochen-lise thanks for reporting the issue. This has been fixed in pre-release version 0.58.x of Java Debugger extension. Pls try the pre-release version.

**Comment by jochen-lise** [comment]
@testforstephen Hello, thank you very much! With version v0.58.2024090204 (pre-release) everything works as expected! :)
